From 26f06420a021ced31446ea1e04af36572e3508f4 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Thu, 13 Sep 2012 16:43:10 -0500
Subject: ENGR00224886 board-mx6q-sabreauto: remove camera control lines

* Remove camera control lines, this gpio control lines are used by
  other periherals
* GPIO_4_5 correspond to RGMII_INT
* GPIO_3_24 correspond to CSI0_DAT5 parallel tv-in

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabreauto.c    |   12 ------------
 arch/arm/mach-mx6/board-mx6q_sabreauto.h    |    4 ----
 arch/arm/mach-mx6/board-mx6solo_sabreauto.h |    4 ----
 3 files changed, 20 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index bb310a9..41e89f1 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -102,7 +102,6 @@
 #define SABREAUTO_DISP0_PWR		IMX_GPIO_NR(3, 24)
 #define SABREAUTO_DISP0_I2C_EN		IMX_GPIO_NR(3, 28)
 #define SABREAUTO_DISP0_DET_INT		IMX_GPIO_NR(3, 31)
-#define SABREAUTO_CSI0_RST		IMX_GPIO_NR(4, 5)
 #define SABREAUTO_DISP0_RESET		IMX_GPIO_NR(5, 0)
 #define SABREAUTO_I2C3_STEER		IMX_GPIO_NR(5, 4)
 #define SABREAUTO_WEIM_NOR_WDOG1        IMX_GPIO_NR(4, 29)
@@ -110,7 +109,6 @@
 #define SABREAUTO_PMIC_INT		IMX_GPIO_NR(5, 16)
 #define SABREAUTO_ALS_INT		IMX_GPIO_NR(5, 17)
 #define SABREAUTO_SD1_WP		IMX_GPIO_NR(5, 20)
-#define SABREAUTO_CSI0_PWN		IMX_GPIO_NR(5, 23)
 #define SABREAUTO_USB_HOST1_OC		IMX_GPIO_NR(5, 0)
 #define SABREAUTO_SD3_CD		IMX_GPIO_NR(6, 15)
 
@@ -1284,16 +1282,6 @@ early_param("can0", early_enable_can0);
 
 static inline void __init mx6q_csi0_io_init(void)
 {
-	/* Camera reset */
-	gpio_request(SABREAUTO_CSI0_RST, "cam-reset");
-	gpio_direction_output(SABREAUTO_CSI0_RST, 1);
-
-	/* Camera power down */
-	gpio_request(SABREAUTO_CSI0_PWN, "cam-pwdn");
-	gpio_direction_output(SABREAUTO_CSI0_PWN, 1);
-	msleep(1);
-	gpio_set_value(SABREAUTO_CSI0_PWN, 0);
-
 	if (cpu_is_mx6q())
 		mxc_iomux_set_gpr_register(1, 19, 1, 1);
 	else if (cpu_is_mx6dl())
diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.h b/arch/arm/mach-mx6/board-mx6q_sabreauto.h
index c087de6..9b5af3c 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.h
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.h
@@ -207,10 +207,6 @@ static iomux_v3_cfg_t mx6q_sabreauto_pads[] = {
 	/* HDMI */
 	MX6Q_PAD_EIM_A25__HDMI_TX_CEC_LINE,
 
-	/* camera reset */
-	MX6Q_PAD_GPIO_19__GPIO_4_5,
-	MX6Q_PAD_EIM_D24__GPIO_3_24,
-
 	/* MLB150 */
 	MX6Q_PAD_ENET_TXD1__MLB_MLBCLK,
 	MX6Q_PAD_GPIO_6__MLB_MLBSIG,
diff --git a/arch/arm/mach-mx6/board-mx6solo_sabreauto.h b/arch/arm/mach-mx6/board-mx6solo_sabreauto.h
index 9111e6b..b752faa 100644
--- a/arch/arm/mach-mx6/board-mx6solo_sabreauto.h
+++ b/arch/arm/mach-mx6/board-mx6solo_sabreauto.h
@@ -207,10 +207,6 @@ static iomux_v3_cfg_t mx6dl_sabreauto_pads[] = {
 	/* HDMI */
 	MX6DL_PAD_EIM_A25__HDMI_TX_CEC_LINE,
 
-	/* camera reset */
-	MX6DL_PAD_GPIO_19__GPIO_4_5,
-	MX6DL_PAD_EIM_D24__GPIO_3_24,
-
 	/* MLB150 */
 	MX6DL_PAD_ENET_TXD1__MLB_MLBCLK,
 	MX6DL_PAD_GPIO_6__MLB_MLBSIG,
-- 
1.7.9.5

